### PR TITLE
Allow single value category features

### PR DESCRIPTION
**Pull request by abidwael**

### CODE REVIEW

**Review by arnavgarg1 — CHANGES_REQUESTED** [review]
Let's log a warning or info block letting the user know that this feature isn't going to add much to the model, and that they should consider dropping it (although it will not actually cause an error per say)

File: ludwig/features/category_feature.py [hunk 1]
```diff
@@ -126,8 +125,8 @@ def get_feature_meta(column, preprocessing_parameters, backend):
         )
         vocab_size = len(str2idx)
         if vocab_size <= 1:
-            raise InputDataError(
-                column.name, CATEGORY, f"At least 2 distinct values are required, column only contains {str(idx2str)}"
+            logger.warning(
+                f"Feature {column.name} of type {CATEGORY} only contains {vocab_size} distinct values: {str(idx2str)}."
```
Comment (arnavgarg1) [inline ludwig/features/category_feature.py:129]: This warning by itself doesn't say much. I'd probably expand and add in that this feature will not be useful for the machine learning model and that they should consider dropping it

**Review by tgaddair — CHANGES_REQUESTED** [review]
There was actually a bug that occurred when this was supported. We should (1) make sure we have a workaround if we're going to support this, and (2) make sure we have a test that verified correctness.

File: ludwig/features/category_feature.py [hunk 2]
```diff
@@ -126,8 +125,9 @@ def get_feature_meta(column, preprocessing_parameters, backend):
         )
         vocab_size = len(str2idx)
         if vocab_size <= 1:
-            raise InputDataError(
-                column.name, CATEGORY, f"At least 2 distinct values are required, column only contains {str(idx2str)}"
+            logger.warning(
+                f"Feature {column.name} of type {CATEGORY} only contains {vocab_size} distinct "
+                "values: {str(idx2str)}. Consider dropping the feature."
```
Comment (arnavgarg1) [inline ludwig/features/category_feature.py:130]: I think you may need to add an `f` before `"values: {str(idx2str)}. Consider dropping the feature."`, but if not, this looks good! Thanks for making the change!

Comment (abidwael) [reply ludwig/features/category_feature.py:130]: Good catch!

**Comment by abidwael** [comment]
Closing since it's more useful to throw an error if a categorical feature with a single distinct value is specified.

**Pull request closed without merging**